src/main/java/com/fasterxml/jackson/databind/deser/impl/ValueInjector.java

```diff
@@ -69,7 +69,14 @@ public void inject(DeserializationContext context, Object beanInstance)
         throws IOException
     {
         final Object value = findValue(context, beanInstance);
-        if (!JacksonInject.Value.empty().equals(value)) {
+
+        if (value == JacksonInject.Value.empty()) {
+            if (Boolean.FALSE.equals(_optional)) {
+                throw context.missingInjectableValueException(String.format(
+                        "No 'injectableValues' configured, cannot inject value with id '%s'", _valueId),
```

Comment:
This is wrong/misleading: there may be (other) `injectableValues`, but not specific injectable value being looked for.